File: backend/internal/models/stock.go
package models

import "time"

type StockMetadata struct {
	Symbol   string `json:"symbol"`
	Name     string `json:"name"`
	Exchange string `json:"exchange"`
	Currency string `json:"currency"`
}

type StockQuote struct {
	Symbol        string    `json:"symbol"`
	Price         float64   `json:"price"`
	Change        float64   `json:"change"`
	ChangePercent float64   `json:"change_percent"`
	Open          float64   `json:"open"`
	High          float64   `json:"high"`
	Low           float64   `json:"low"`
	PreviousClose float64   `json:"previous_close"`
	Currency      string    `json:"currency"`
	MarketStatus  string    `json:"market_status"`
	UpdatedAt     time.Time `json:"updated_at"`
	Cached        bool      `json:"cached"`
}

type StockPage struct {
	Stocks []StockMetadata
	Page   int
	Limit  int
	Total  int
}

type StockQuoteDetail struct {
	Symbol            string  `json:"symbol"`
	Name              string  `json:"name"`
	Price             float64 `json:"price"`
	Change            float64 `json:"change"`
	ChangePercent     float64 `json:"change_percent"`
	Open              float64 `json:"open"`
	High              float64 `json:"high"`
	Low               float64 `json:"low"`
	PreviousClose     float64 `json:"previous_close"`
	Currency          string  `json:"currency"`
	MarketStatus      string  `json:"market_status"`
	MarketStatusUntil string  `json:"market_status_until"`
	UpdatedAt         string  `json:"updated_at"`
	Cached            bool    `json:"cached"`
}

type QuoteError struct {
	Symbol  string `json:"symbol"`
	Code    string `json:"code"`
	Message string `json:"message"`
}

type MultipleQuotesResult struct {
	Quotes []StockQuote `json:"quotes"`
	Errors []QuoteError `json:"errors"`
}

type StockCandle struct {
	Timestamp time.Time `json:"timestamp"`
	Open      float64   `json:"open"`
	High      float64   `json:"high"`
	Low       float64   `json:"low"`
	Close     float64   `json:"close"`
	Volume    float64   `json:"volume"`
}

type StockCandleSeries struct {
	Symbol   string        `json:"symbol"`
	Range    string        `json:"range"`
	Interval string        `json:"interval"`
	Currency string        `json:"currency"`
	Timezone string        `json:"timezone"`
	Candles  []StockCandle `json:"candles"`
	Cached   bool          `json:"cached"`
}

File: frontend/src/features/stocks/types/stock.ts
export interface Stock {
  symbol: string;
  name: string;
  exchange: string;
  currency: string;
}

export interface StockQuote extends Stock {
  price: number;
  change: number;
  change_percent: number;
  open: number;
  high: number;
  low: number;
  previous_close: number;
  market_status: string;
  market_status_until: string;
  updated_at: string;
  cached: boolean;
}

export type CandleRange = "1D" | "1W" | "1M" | "3M" | "1Y" | "MAX";
export interface StockCandle { timestamp: string; open: number; high: number; low: number; close: number; volume: number; }
export interface StockCandleSeries {
  symbol: string; range: CandleRange; interval: string; currency: string; timezone: string;
  candles: StockCandle[]; cached: boolean;
}
